--- upstream/tinyfiledialogs.h	2019-10-23 13:14:06.000000000 -0500
+++ cbits/tinyfiledialogs.h	2019-10-23 13:23:00.000000000 -0500
@@ -192,6 +192,8 @@
 		/* aDefaultRGB and aoResultRGB can be the same array */
 		/* returns NULL on cancel */
 
+// MT: removing these since we're not using them and c2hs has trouble parsing
+#if 0
 
 /************ NOT CROSS PLATFORM SECTION STARTS HERE ************************/
 #ifdef _WIN32
@@ -271,6 +273,8 @@
 
 #endif /*_WIN32 */
 
+#endif // MT: end removed section
+
 #ifdef	__cplusplus
 }
 #endif
